generator/internal/stub_generator_base.cc

```diff
@@ -0,0 +1,148 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "generator/internal/stub_generator_base.h"
+#include "generator/internal/predicate_utils.h"
+#include <google/protobuf/descriptor.h>
+
+namespace google {
+namespace cloud {
+namespace generator_internal {
+
+StubGeneratorBase::StubGeneratorBase(
+    std::string const& header_path_key, std::string const& cc_path_key,
+    google::protobuf::ServiceDescriptor const* service_descriptor,
+    VarsDictionary service_vars,
+    std::map<std::string, VarsDictionary> service_method_vars,
+    google::protobuf::compiler::GeneratorContext* context)
+    : ServiceCodeGenerator(header_path_key, cc_path_key, service_descriptor,
+                           std::move(service_vars),
+                           std::move(service_method_vars), context) {}
+
+void StubGeneratorBase::HeaderPrintPublicMethods() {
+  for (auto const& method : methods()) {
+    if (IsStreamingWrite(method)) {
+      HeaderPrintMethod(method, __FILE__, __LINE__, R"""(
+  std::unique_ptr<::google::cloud::internal::StreamingWriteRpc<
```

Comment:
I think I see that this is preserving previous behavior, but the rule of thumb is only to fully qualify a name in a top-level using declaration, or where it might be required.  So, I'd vote for dropping the leading `::` here and elsewhere.

Reply:
Ack. I could do that in a follow up PR

Reply:
SGTM.